### ISSUE TITLE
how to use another api base 

### ISSUE THREAD

**Issue by lihan0421** [SWE-agent/SWE-agent#869]
### Describe the issue

i created .env
# Uncomment the line below and add your OpenAI API Key
OPENAI_API_KEY='sk-7NqIs9uG8ILNQw....'
api_key ='sk-7NqIs9uG8ILNQw56RCk....'
api_base='https://xiaoai.plus'
OPENAI_API_BASE_URL=' https://xiaoai.plus'

but it cant work

**Comment by lwang2070** [comment]
Which version are you using? In the latest version, they use litellm to call openai models, which strangely uses the following environment variable:

```
api_base = (
                api_base  # for deepinfra/perplexity/anyscale/groq/friendliai we check in get_llm_provider and pass in the api base from there
                or litellm.api_base
                or get_secret("OPENAI_API_BASE")
                or "https://api.openai.com/v1"
            )
```

Try using `OPENAI_API_BASE` to set the base url.

**Comment by klieret** [comment]
Thanks @lwang2070 , that should indeed be the way to do it!

**Comment by d10r** [comment]
setting `OPENAI_API_BASE` and `OPENAI_API_KEY` didn't fix it for me.
I always get something like:
```
$ sweagent run --agent.model.name=openai/llama-3.3-70b --env.repo.github_url=https://github.com/SWE-agent/test-repo --problem_statement.github_url=https://github.com/SWE-agent/test-repo/issues/1
...
  File "/home/didi/dev-tools/SWE-agent/sweagent/agent/models.py", line 548, in __init__
    raise ValueError(msg)
ValueError: Model openai/llama-3.3-70b does not support function calling
```

The specific error is likely a red herring. I get the same if setting an invalid API key or API url.
I added the _openai_ prefix to the model name because that's how litellm knows what kind of endpoint to use according to https://docs.litellm.ai/docs/providers/openai_compatible. (but also tried without that prefix, same error)

I tried with the venice.ai API and with an ollama API. Both are OpenAI compatible and work with other tools (e.g. aichat) using the openai API.

**Comment by klieret** [comment]
Hi! This is not related to the model itself but to the way we parse the actions from the model. The default currently is using function calling and not all models support that. To change that set `parse_function` to `thought_action`, see here: https://swe-agent.com/latest/faq/. 

Does that help?

**Comment by klieret** [comment]
Also going back to the original question, the best way is to just set the base per config or per command line option `--agent.model.api_base=...` instead of with environment variables (which _should_ work, but not sure)

**Comment by d10r** [comment]
Thanks für the feedback, now I understand a bit better.
I got past this error, now getting this:
```
$ sweagent run --config config/default_no_fcalls.yaml --agent.model.api_base=https://api.venice.ai/api/v1 --agent.model.name=llama-3.3-70b --env.repo.github_url=https://github.com/SWE-agent/test-repo --problem_statement.github_url=https://github.com/SWE-agent/test-repo/issues/1
👋 INFO     This is SWE-agent version 1.0.0 (ba555859d1ffa681a473659bf109f0879d4e8684) with SWE-ReX 1.1.0 ().                                                                                                              
🔧 INFO     Setting problem statement based on github issue url                                                                                                                                                            
🔧 INFO     Loaded environment variables from /home/didi/dev-tools/SWE-agent/.env                                                                                                                                          
Traceback (most recent call last):
  File "/home/didi/.local/share/virtualenvs/SWE-agent-4hFPuGIZ/bin/sweagent", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/didi/dev-tools/SWE-agent/sweagent/run/run.py", line 87, in main
    run_single_main(remaining_args)
  File "/home/didi/dev-tools/SWE-agent/sweagent/run/run_single.py", line 212, in run_from_cli
    run_from_config(BasicCLI(RunSingleConfig, help_text=help_text).get_config(args))  # type: ignore
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/didi/dev-tools/SWE-agent/sweagent/run/run_single.py", line 202, in run_from_config
    RunSingle.from_config(config).run()
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/didi/dev-tools/SWE-agent/sweagent/run/run_single.py", line 160, in from_config
    agent = get_agent_from_config(config.agent)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/didi/dev-tools/SWE-agent/sweagent/agent/agents.py", line 206, in get_agent_from_config
    return DefaultAgent.from_config(config)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/didi/dev-tools/SWE-agent/sweagent/agent/agents.py", line 458, in from_config
    model = get_model(config.model, config.tools)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/didi/dev-tools/SWE-agent/sweagent/agent/models.py", line 779, in get_model
    return LiteLLMModel(args, tools)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/didi/dev-tools/SWE-agent/sweagent/agent/models.py", line 551, in __init__
    self.lm_provider = litellm.model_cost[self.config.name]["litellm_provider"]
                       ~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
KeyError: 'llama-3.3-70b'
```

(also tried with "openai/llama-3.3-70b" - same result)

**Comment by klieret** [comment]
@d10r You're right. Once #937 is merged, you can pull and try again, hopefully that will solve it.

**Comment by d10r** [comment]
can confirm, working now. Thank you!

**Comment by klieret** [comment]
Thanks for confirming, great to hear! (if something else comes up, please open a new issue, as comments on old ones might go unnoticed)
